src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/impl/MicrosoftOAuth2AP.java

```diff
@@ -0,0 +1,60 @@
+package edu.harvard.iq.dataverse.authorization.providers.oauth2.impl;
+
+import com.github.scribejava.apis.MicrosoftAzureActiveDirectory20Api;
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import edu.harvard.iq.dataverse.authorization.providers.oauth2.AbstractOAuth2AuthenticationProvider;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.logging.Logger;
+import java.io.StringReader;
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import edu.harvard.iq.dataverse.authorization.AuthenticatedUserDisplayInfo;
+
+/**
+ *
+ * @author
+ */
+public class MicrosoftOAuth2AP extends AbstractOAuth2AuthenticationProvider{
+
+    private static final Logger logger = Logger.getLogger(MicrosoftOAuth2AP.class.getCanonicalName());
+
+    public MicrosoftOAuth2AP(String aClientId, String aClientSecret){
+        this.id = "microsoft";
+        this.title = "Microsoft";
+        this.clientId = aClientId;
+        this.clientSecret = aClientSecret;
+        this.scope = Arrays.asList("openid User.Read");
+        this.baseUserEndpoint = "https://graph.microsoft.com/v1.0/me";
+    }
+
+    @Override
+    public DefaultApi20 getApiInstance(){
+        return MicrosoftAzureActiveDirectory20Api.instance();
+    }
+
+    @Override
+    protected ParsedUserResponse parseUserResponse(final String responseBody) {
+        try ( StringReader rdr = new StringReader(responseBody);
+              JsonReader jrdr = Json.createReader(rdr) )  {
+            JsonObject response = jrdr.readObject();
+            AuthenticatedUserDisplayInfo displayInfo = new AuthenticatedUserDisplayInfo(
+                    response.getString("givenName", ""),
+                    response.getString("surname", ""),
+                    response.getString("userPrincipalName", ""),
+                    "", "");
```

Comment:
This might be beyond scope, but it would be totally awesome trying to receive values for `affiliation` fields.

- https://docs.microsoft.com/en-us/graph/api/resources/organization?view=graph-rest-1.0
- https://docs.microsoft.com/en-us/graph/api/organization-get?view=graph-rest-1.0
- https://docs.microsoft.com/en-us/graph/permissions-reference#organization-permissions

Reply:
We view the options for receive these values and can implement this in the future. It's really interesting for us. :D